streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateTaskTest.java

```diff
@@ -127,7 +128,7 @@ public void shouldInitializeProcessorTopology() {
     @Test
     public void shouldProcessRecordsForTopic() {
         globalStateTask.initialize();
-        globalStateTask.update(new ConsumerRecord<>(topic1, 1, 1, "foo".getBytes(), "bar".getBytes()));
+        globalStateTask.update(record(topic1, 1, 1, "foo".getBytes(), "bar".getBytes()));
```

Comment:
That constructor of ConsumerRecord set the timestamp to `-1`, which is now prohibited because we construct a Record before processing, and Record enforces no negative timestamps.

This seems fine to me, since it would only happen in unit tests (as ConsumerRecords returned from the broker never have negative timestamps).